Feat/회원가입 기능 구현 #8

haechangchoi opened the pull request:
## 🔧 어떤 기능인가요?

> 회원 가입 기능 서비스 및 컨트롤러 추가

## #️⃣ 연관된 이슈

> #8

## 💡 리뷰어에게 하고 싶은 말
- [x] 회원 가입 기능이 정상적으로 동작하는지 확인
- [x] 예외 처리 로직에 대한 개선점
- [x] MemberService 및 MemberController에 대한 구조적인 피드백

<!-- 코드에 나타나지 않는 설계 및 의도, 중점적으로 봐줬으면 하는 점, 특이한 변경사항 등 -->

## 🙏 아래 내용이 모두 충족 되었는지 확인해주세요 🙏

- [x] PR 이전 `dev` 브랜치 병합 하셨나요?
- [x] PR 이전 빌드 테스트 정상적으로 성공했나요?
- [x] PR 상세내용이 충분히 기재 되었나요?
- [x] PR 리뷰어, 할당자, 라벨, 프로젝트 확인

haechangchoi committed:
✨ [feat] 회원 관련 엔티티 및 레포지토리 추가

haechangchoi committed:
🛠 [build] ModelMapper 의존성 추가

haechangchoi committed:
♻️[refactor] 회원 관련 엔티티 및 레포지토리 리팩토링

haechangchoi committed:
✨ [feat] 에러 코드 추가 및 예외 처리 개선

haechangchoi committed:
✨ [feat] 회원 가입 기능 서비스 및 컨트롤러 추가
✨ [feat] 회원 가입 API 엔드포인트 구현
✅ [test] 회원 가입 기능 테스트 코드 추가

kendrickhama commented on src/main/java/com/PickOne/domain/user/controller/MemberController.java:
```diff
@@ -0,0 +1,26 @@
+
+    private final MemberService memberService;
+
+    @PostMapping
+    public ResponseEntity<BaseResponse<Void>> create(@RequestBody MemberCreateDto dto) {
+        memberService.createMember(dto);
+        return BaseResponse.success();
+    }
```
- [ ] 매서드명은 create -> signUp 어떨까요
- [ ] BaseResponse.success **(SuccessCode.CREATED)**  로 부탁드립니다! (성공코드 반환)

kendrickhama reviewed (commented):
리뷰 완료했습니다

haechangchoi committed:
♻️[refactor] 메서드명 create를 signUp으로 변경 및 성공 응답 수정

- 메서드명을 `create`에서 `signUp`으로 변경하여 의미를 명확히 함.
- 성공 응답을 `BaseResponse.success(SuccessCode.CREATED)`로 수정.
